### ISSUE TITLE
 import high voltage line

### ISSUE THREAD

**Issue by solido3d** [vvoovv/blender-geo#207]
Hi . I would like to be able to import high voltage line too. it would be possible.
Thank you

**Comment by vvoovv** [comment]
Download the file [power_lines_and_towers.py](https://github.com/vvoovv/blender-osm/blob/setup_scripts/power_lines_and_towers.py). 
Remove the lines 60-69. They are for the power towers

Set the field "Setup script" to that file and import in the "3D simple" mode.

**Comment by solido3d** [comment]
Hello, I understand that I have to modify the downloaded file. but I don't understand where to put it next. I do not know where it is"Setup script"
thanks

**Comment by vvoovv** [comment]
The option "Setup script" is located in the addon panel (in the right side panel of the Blender's 3D View).

**Comment by solido3d** [comment]
solved. Thank you very much. It would be possible to import the rivers

**Comment by vvoovv** [comment]
Rivers are imported as areas by default.

**Comment by vvoovv** [comment]
To import rivers as Blender curves:

Create a copy of your file _power_lines_and_towers.py_:
Change the line 55 to:
```
        lambda tags, e: tags.get("waterway") == "river",
```
Save the new file and use it in the field _Setup script_.

**Comment by solido3d** [comment]
Thanks .Perfect

**Comment by solido3d** [comment]
How to import the electrical lines and that they are imported with the vertices of the towers only.
Thank yo
![powerPoint](https://user-images.githubusercontent.com/8347837/166892824-cf8448db-e452-4cb0-a644-46bf33551353.jpg)
u

**Comment by vvoovv** [comment]
You need a 3D model of a tower.

Use the setup script [power_lines_and_towers.py](https://github.com/vvoovv/blender-osm/blob/setup_scripts/power_lines_and_towers.py). Edit the line [68](https://github.com/vvoovv/blender-osm/blob/setup_scripts/power_lines_and_towers.py#L68) to provide the path, the file name and the collection name for the 3D model of the tower.

**Comment by solido3d** [comment]
no, what I need is that the line only has vertices where the tower is
Thank you.

**Comment by vvoovv** [comment]
> no, what I need is that the line only has vertices where the tower is Thank you.

I described that [here](#issuecomment-667509713)

**Comment by vvoovv** [comment]
Sorry, I didn't understand what you meant.

Replace _wayManager_ for _linestring_ in the line 57.

**Comment by solido3d** [comment]
Thank you so much

**Comment by Wickid22** [comment]
"Replace wayManager for linestring"

This does not work for me. I still get the powerline as in the screenshot above. I tried changing the tag to "tower" instead of "line" in line 55

https://github.com/vvoovv/blender-osm/blob/72acbef91cb37476b65bf9d2237dc4711fe4a0e4/power_lines_and_towers.py#L55 

But this imports nothing to my blender file. How do I get a Bezier curve with only the tower location as nodes?

**Comment by vvoovv** [comment]
Hi @Wickid22 

You can get power lines without intermediary points if you leave the field "Terrain" empty. Then you can project the points on the terrain in your Geometry Nodes-setup using the Raycast node.

**Comment by Wickid22** [comment]
@vvoovv 

Is there also a way to get windturbine locations using this script? I tried:

![image](https://user-images.githubusercontent.com/120053546/212543996-54c6c0f9-6f47-4ffa-b444-5459a75ddea3.png)

However, that only gives me photovoltaic units. Even though the windturbines are also tagged with "power" and "generator". My guess is that it is due to a windturbine being a node, but I can't figure out how to rewrite this part of the script.

**Comment by vvoovv** [comment]
Wind turbines are point objects in OSM.

The related lines:
```python
    osm.addNodeCondition(
        lambda tags, e: tags.get("generator:source") == "wind",
        "trees",
        None,
        BaseNodeRenderer(app, path, filename, collection)
    )
```

**Comment by Wickid22** [comment]
Hi, 

Thankyou for the quick reply! I've been trying for a bit but the script fails on:

`NameError: name 'path' is not defined`

I tried googling this and I found:

`from pathlib import Path`

But I don't think this is what I should be doing. In any case, adding the above makes it fail on 

`NameError: name 'filename' is not defined`

So I'm pretty sure I'm doing something wrong, but my Python knowledge is limited at best.

**Comment by vvoovv** [comment]
`path` is the full folder name where the Blender file with the name `filename` is located.
That file must contain the `collection`.
